codegen/template.go

```diff
@@ -81,9 +99,17 @@ type Template struct {
 
 // NewTemplate creates a bundle of templates.
 func NewTemplate() (*Template, error) {
+	return NewTemplateFromAssetProvider(&defaultAssetCollection{})
+}
+
+// NewTemplateFromAssetProvider returns a template helper for the
+// provided asset collection
+func NewTemplateFromAssetProvider(
+	assetProvider AssetProvider,
+) (*Template, error) {
 	t := tmpl.New("main").Funcs(funcMap)
```

Comment:
`funcMap` is not exposed here. If we want to use customized template, `funcMap` should also be part of the params of constructor `NewTemplateFromAssetProvider`.